integration-libs/opf/checkout/components/b2b/opf-b2b-checkout-payment-type/opf-b2b-checkout-payment-type.component.ts

```diff
@@ -0,0 +1,206 @@
+      })
+    );
+
+  typeSelected$: Observable<PaymentType> = combineLatest([
+    this.checkoutPaymentTypeFacade.getSelectedPaymentTypeState().pipe(
+      filter((state) => !state.loading),
+      map((state) => state.data)
+    ),
+    this.paymentTypes$,
+  ]).pipe(
+    map(
+      ([selectedPaymentType, availablePaymentTypes]: [
+        PaymentType | undefined,
+        PaymentType[],
+      ]) => {
+        if (
+          selectedPaymentType &&
+          availablePaymentTypes.find((availablePaymentType) => {
+            return availablePaymentType.code === selectedPaymentType.code;
+          })
+        ) {
+          return selectedPaymentType;
+        }
+        if (availablePaymentTypes.length) {
+          this.busy$.next(true);
+          this.checkoutPaymentTypeFacade
+            .setPaymentType(
+              availablePaymentTypes[0].code as string,
+              this.poNumberInputElement?.nativeElement?.value
+            )
+            .subscribe({
+              complete: () => this.onSuccess(),
+              error: () => this.onError(),
+            });
+          return availablePaymentTypes[0];
+        }
+        return undefined;
+      }
+    ),
+    filter(isNotUndefined),
+    distinctUntilChanged(),
+    tap((selected) => {
+      this.typeSelected = selected?.code;
+      this.checkoutStepService.disableEnableStep(
+        CheckoutStepType.PAYMENT_DETAILS,
+        selected?.code === B2BPaymentTypeEnum.ACCOUNT_PAYMENT
+      );
+      this.checkoutStepService.disableEnableStep(
+        CheckoutStepType.REVIEW_ORDER,
+        selected?.code === B2BPaymentTypeEnum.CARD_PAYMENT
+      );
+    })
+  );
+
+  cartPoNumber$: Observable<string> = this.checkoutPaymentTypeFacade
+    .getPurchaseOrderNumberState()
+    .pipe(
+      filter((state) => !state.loading),
+      map((state) => state.data),
+      filter(isNotUndefined),
+      distinctUntilChanged()
+    );
+
+  constructor(
+    protected checkoutPaymentTypeFacade: CheckoutPaymentTypeFacade,
+    protected checkoutStepService: CheckoutStepService,
+    protected activatedRoute: ActivatedRoute,
+    protected globalMessageService: GlobalMessageService
+  ) {}
+
+  changeType(code: string): void {
+    this.busy$.next(true);
+    this.typeSelected = code;
+
+    this.checkoutPaymentTypeFacade
+      .setPaymentType(code, this.poNumberInputElement?.nativeElement.value)
+      .subscribe({
+        complete: () => this.onSuccess(),
+        error: () => this.onError(),
+      });
+  }
+
+  next(): void {
+    if (!this.typeSelected) {
+      return;
+    }
+
+    const poNumberInput = this.poNumberInputElement?.nativeElement.value;
+    // if the PO number didn't change
+    if (poNumberInput === getLastValueSync(this.cartPoNumber$)) {
+      this.checkoutStepService.next(this.activatedRoute);
+      return;
+    }
+
+    this.busy$.next(true);
+    this.checkoutPaymentTypeFacade
+      .setPaymentType(this.typeSelected, poNumberInput)
```

Comment:
do we need to unsubscribe or we are 100% sure it complete after 1 time?
In the doubt should we add 'take(1)' or sub.add + ngDestroy callback?